src/App.ts

```diff
@@ -37,11 +37,11 @@ interface AppOptions {
   scriptHash?: string,
 }
 
-type OnRouteCallback = (arg0: Page, arg1: Page, model: Model, params: any, done?: () => void) => void;
+type OnRouteCallback<T = object> = (arg0: Page, arg1: Page, model: ChildModel<T>, params: PageParams, done?: () => void) => void;
```

Comment:
I suggest using just `Model` here, as the routes receive `page.model`, which is a root model - instance of the base class `Model` in racer.

Most other places should be ok with just `Model` too since Derby doesn't really use the internal  underscore-prefixed ChildModel fields.